SwrveSDK/src/main/java/com/swrve/sdk/SwrveImp.java

```diff
@@ -1278,11 +1276,29 @@ protected void initCampaigns() {
         }
     }
 
+    private void loadCampaignsStateFromCache() throws JSONException {
+        String campaignsStateFromCache = cachedLocalStorage.getCacheEntryForUser(userId, CAMPAIGNS_STATE_CATEGORY);
+        if (!SwrveHelper.isNullOrEmpty(campaignsStateFromCache)) {
+            JSONObject campaignsStateJson = new JSONObject(campaignsStateFromCache);
+            Iterator<String> campaignIdIterator = campaignsStateJson.keys();
+            while(campaignIdIterator.hasNext()) {
+                String campaignIdStr = campaignIdIterator.next();
+                try {
+                    int campaignId = Integer.parseInt(campaignIdStr);
+                    SwrveCampaignState campaignState = new SwrveCampaignState(campaignsStateJson.getJSONObject(campaignIdStr));
+                    campaignsState.put(campaignId, campaignState);
+                } catch(Exception exp) {
+                    SwrveLogger.e(LOG_TAG, "Could not load state for campaign " + campaignIdStr, exp);
```

Comment:
If there's ever corrupt data saved to the local cache, do we recover ok from this?